script/licenses.py

```diff
@@ -244,8 +242,88 @@ def main(argv: Sequence[str] | None = None) -> int:
     return exit_code
 
 
+def extract_license_classifier(classifiers: list[str] | None) -> list[str]:
+    """Extract license from list of classifiers."""
+    return [
+        license_classifier
+        for classifier in classifiers or ()
+        if classifier.startswith("License")
+        and (license_classifier := classifier.rpartition(" :: ")[2])
+        and license_classifier != "OSI Approved"
+    ]
+
+
+def get_package_metadata(dist: metadata.Distribution) -> PackageMetadata:
+    """Get package metadata for distribution."""
+    return {
+        "name": dist.name,
+        "version": dist.version,
+        "license_expression": dist.metadata.get("license-expression"),
+        "license_metadata": dist.metadata.get("license"),
+        "license_classifier": extract_license_classifier(
+            dist.metadata.get_all("classifier")
+        ),
+    }
+
+
+def extract_licenses(args: ExtractArgs) -> int:
+    """Extract license data for installed packages."""
+    licenses = sorted(
+        [get_package_metadata(dist) for dist in list(metadata.distributions())],
+        key=lambda dist: dist["name"],
+    )
+    Path(args.output_file).write_text(json.dumps(licenses, indent=2))
+    return 0
```

Comment:
This doesn't seem needed if we never return another integer.

Reply:
I'd like to keep that as I consider it the "entry point" to the subcommand. The returns of both functions `extract_licenses` and `check_licenses` are directly returned from `main` and passed to `sys.exit`.

Reply:
Ok.